modules/nw-cudn-cr.adoc

```diff
@@ -0,0 +1,146 @@
+        - cidr: 10.100.0.0/16
+----
+<1> Name of your `UserDefinedNetwork` resource. This should not be `default` or duplicate any global namespaces created by the Cluster Network Operator (CNO).
+<2> Specifies the label key to match.
+<3> Specifies the operator. Valid values include: `In`, `NotIn`, `Exists`, and `DoesNotExist`.
+<4> Specifies an array of string values. If the `operator` value is either `Exists` or `DoesNotExist`, the value array must be empty.
+<5> The `topology` field describes the network configuration; accepted values are `Layer2` and `Layer3`. Specifying a `Layer3` topology type creates a layer 2 segment per node, each with a different subnet. Layer 3 routing is used to interconnect node subnets.
+<6> This field specifies the topology configuration. Valid values are `layer2` or `layer3`.
+<7> Specifies a `Primary` or `Secondary` role. `Primary` is the only `role` specification supported in {product-version}.
+<8> For `Layer3` topology types the following specifies config details for the `subnet` field:
++
+* The `subnets` field is mandatory.
+* The type for the `subnets` field is `cidr` and `hostSubnet`:
+** `cidr` is the cluster subnet and accepts a string value.
+** `hostSubnet` specifies the nodes subnet prefix that the cluster subnet is split to.
+** For IPv6, only a `/64` length is supported for `hostSubnet`.
++
+. Apply your request by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f <my_layer_two_udn.yaml>
+----
++
+Where `<my_layer_two_udn.yaml>` is the name of your `Layer2` or `Layer3` configuration file.
+
+. Verify that your request is successful by running the following command:
++
+[source, terminal]
+----
+$ oc get clusteruserdefinednetwork <cudn_name> -o yaml
+----
++
+Where `<cudn_name>` is the name you created of your cluster-wide user-defined network.
++
+.Example output
+[source,terminal]
+----
+apiVersion: k8s.ovn.org/v1
+kind: ClusterUserDefinedNetwork
+metadata:
+  annotations:
+    kubectl.kubernetes.io/last-applied-configuration: |
+      {"apiVersion":"k8s.ovn.org/v1","kind":"ClusterUserDefinedNetwork","metadata":{"annotations":{},"name":"my-cudn"},"spec":{"namespaceSelector":{"matchExpressions":[{"key":"custom.network.selector","operator":"In","values":["example-namespace-1","example-namespace-2","example-namespace-3"]}]},"network":{"layer3":{"role":"Primary","subnets":[{"cidr":"10.100.0.0/16"}]},"topology":"Layer3"}}}
+  creationTimestamp: "2024-11-19T16:46:34Z"
+  finalizers:
+  - k8s.ovn.org/user-defined-network-protection
+  generation: 1
+  name: my-cudn
+  resourceVersion: "47985"
+  uid: 16ee0fcf-74d1-4826-a6b7-25c737c1a634
+spec:
+  namespaceSelector:
+    matchExpressions:
+    - key: custom.network.selector
+      operator: In
+      values:
+      - example-namespace-1
+      - example-namespace-2
+      - example-namespace-3
+  network:
+    layer3:
+      role: Primary
+      subnets:
+      - cidr: 10.100.0.0/16
+    topology: Layer3
+status:
```

Comment:
@tssurya @npinaeva should we update docs with the new NetworkAllocated condition? (on a follow-up PR)?

Reply:
updating the docs is good because that status conditional has already merged so it should be part of this PR